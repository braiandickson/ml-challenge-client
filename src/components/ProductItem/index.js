import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import { formatProductRoute, formatPrice } from '../../utils/formats';

const ProductItem = ({ id, title, price, condition, picture, location }) => {
  return (
    <Link key={id} to={formatProductRoute(id)}>
      <div className="row middle product-item">
        <img className="product-item__picture" src={picture} alt={title} />
        <div className="column product-item__information">
          <div className="row middle product-item__price-container">
            <h2 className="product-item__price">{formatPrice(price)}</h2>
            {condition && <div className="product-item__status" />}
          </div>
          <p className="product-item__title">{title}</p>
        </div>
        <div className="column">
          <span className="product-item__location">{location}</span>
        </div>
      </div>
    </Link>
  );
};

ProductItem.defaultProps = {
  id: '',
  title: '',
  price: {
    currency: '$',
    amount: 0,
    decimals: 0,
  },
  condition: 'Nuevo',
  picture: '',
};

ProductItem.propTypes = {
  id: PropTypes.string,
  title: PropTypes.string,
  price: PropTypes.shape({
    currency: PropTypes.string,
    amount: PropTypes.number,
    decimals: PropTypes.number,
  }),
  condition: PropTypes.string,
  picture: PropTypes.string,
};

export default ProductItem;
