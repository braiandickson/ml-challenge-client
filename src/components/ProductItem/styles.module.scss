@import '../../styles/variables/colors';

.product-item {
  background-color: $white;
  border-bottom: 1px solid $light-grey;
  height: 220px;
  padding: 24px;
  width: 850px;

  &__picture {
    height: 160px;
    margin-right: 24px;
    width: 160px;
  }

  &__information {
    height: 100%;
    width: 100%;
  }

  &__price-container {
    margin-bottom: 10px;
    padding: 10px 0;
  }

  &__price {
    color: $dark-grey;
    font-size: 22px;
  }

  &__status {
    background-color: $green;
    border-radius: 25px;
    height: 15px;
    margin-left: 10px;
    width: 15px;
  }

  &__title {
    color: $dark-grey;
    font-size: 18px;
    max-width: 400px;
  }
}
