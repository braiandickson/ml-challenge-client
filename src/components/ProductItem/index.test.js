import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import ProductItem from './index';

const data = {
  id: 'MLA826445412',
  title: 'Mouse InalÃ¡mbrico Recargable Negro Tedge',
  price: {
    currency: '$',
    amount: 449,
    decimals: 0,
  },
  picture: 'http://mla-s1-p.mlstatic.com/744481-MLA41024544993_032020-I.jpg',
  condition: 'Nuevo',
};

describe('#ProductItem', () => {
  it('Mounts the ProductItem component correctly with data', () => {
    const component = mount(
      <Router>
        <ProductItem {...data} />
      </Router>
    );
    expect(toJson(component)).toMatchSnapshot();
  });
});
