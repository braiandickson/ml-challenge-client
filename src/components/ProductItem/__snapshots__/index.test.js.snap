// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#ProductItem Mounts the ProductItem component correctly with data 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <ProductItem
      condition="Nuevo"
      id="MLA826445412"
      picture="http://mla-s1-p.mlstatic.com/744481-MLA41024544993_032020-I.jpg"
      price={
        Object {
          "amount": 449,
          "currency": "$",
          "decimals": 0,
        }
      }
      title="Mouse Inalámbrico Recargable Negro Tedge"
    >
      <Link
        key="MLA826445412"
        to="/items/MLA826445412"
      >
        <LinkAnchor
          href="/items/MLA826445412"
          navigate={[Function]}
        >
          <a
            href="/items/MLA826445412"
            onClick={[Function]}
          >
            <div
              className="row middle product-item"
            >
              <img
                alt="Mouse Inalámbrico Recargable Negro Tedge"
                className="product-item__picture"
                src="http://mla-s1-p.mlstatic.com/744481-MLA41024544993_032020-I.jpg"
              />
              <div
                className="column product-item__information"
              >
                <div
                  className="row middle product-item__price-container"
                >
                  <h2
                    className="product-item__price"
                  >
                    $449
                  </h2>
                  <div
                    className="product-item__status"
                  />
                </div>
                <p
                  className="product-item__title"
                >
                  Mouse Inalámbrico Recargable Negro Tedge
                </p>
              </div>
              <div
                className="column"
              >
                <span
                  className="product-item__location"
                />
              </div>
            </div>
          </a>
        </LinkAnchor>
      </Link>
    </ProductItem>
  </Router>
</BrowserRouter>
`;
